Add 2.15 docs for authenticated private registries (#1235)
rancher/rancher-product-docs: versions/v2.15/modules/en/pages/rancher-admin/global-configuration/global-default-private-registry.adoc

For instructions on how to set up a private registry during Rancher installation, refer to the {xref-air-gap}[air-gapped installation guide].
====

> A private registry is a private, consistent, and centralized source of truth for the container images in your clusters. You can use a private container image registry to share custom images within your organization, or to prevent rate limiting issues encountered when pulling images from public registries.

Reviewer: ```suggestion
A private registry provides a consistent, centralized source of truth for container images across your clusters. You can use a private registry to share custom images within your organization or to avoid rate limits when pulling from public registries.
```

There are two main ways to set up a private registry in Rancher:

* Set up a _global_ system default registry through the **Settings** tab in the global view.
* Set up a _cluster level_ system default private registry in the advanced options when provisioning or updating a downstream cluster.

> The global system default registry is intended to be used in air-gapped setups, or other environments where all clusters should share the same default registry host. Rancher will use this registry to pull core images required for system-components, and will configure downstream clusters to do the same by default.

Reviewer: ```suggestion
* Set up a global system default registry in the **Settings** tab in the global view.
* Set up a cluster-level system default private registry in **Advanced Options** when provisioning or updating a downstream cluster.

Use the global system default registry in air-gapped environments or other setups where all clusters share the same default registry host. Rancher uses this registry to pull core images required for system components and configures downstream clusters to do the same by default.
```

> Cluster level default registries are intended to be used in scenarios where a specific downstream environment should access a different registry than the global default, or to access the same registry using different credentials.

Reviewer: ```suggestion
Use a cluster-level default registry when a downstream cluster needs a different registry than the global default or requires different credentials for the same registry.
```

[NOTE]
====
For details on how Rancher works with authenticated default registries, see {xref-authenticated-private-registries}[Authenticated Private Registries].
====

[IMPORTANT]
====
Reconfiguring registry details globally can trigger the redeployment of system components in the local cluster and downstream clusters that do not define cluster-level registry configurations. This can temporarily prevent access to downstream clusters through the Rancher UI.
====

[#_configure_a_cluster_level_private_registry]
== Configure a Cluster Level Private Registry

In addition to global registry configurations, Rancher supports cluster-scoped registry configurations. Specify registry details in *Advanced Options* when creating or updating a cluster.

[IMPORTANT]
====
Reconfiguring registry details on an existing cluster can trigger the redeployment of system components in that cluster. This can temporarily prevent access to the cluster through the Rancher UI.
====

[#_using_a_cluster_level_private_registry_with_provisioned_clusters]
=== Using a Cluster Level Private Registry with Provisioned Clusters

. Select *☰ > Cluster Management*.

> . On the *Clusters** page, click *Create*.

Reviewer: ```suggestion
. On the *Clusters* page, click *Create*.
```

*Result:* Rancher pulls system images from your private registry.

[#_set_an_authenticated_private_registry_as_the_default_registry]
== Set an Authenticated Private Registry as the Default Registry

If the desired registry requires authentication (i.e. `.dockerconfigjson`),

. Log into Rancher and configure the default administrator password.
. Using `kubectl`, `terraform`, or the Rancher UI, create a new registry secret in the `cattle-system` namespace
. If the secret is created outside of the Rancher UI, ensure that the appropriate xref:#_working_with_private_registry_credentials[backup labels are set]
. Select **☰ > Global Settings**.
. Go to `system-default-registry-pull-secrets` and choose **⋮ > Edit Setting**.
. Enter the name of the registry pull secret within the `cattle-system` namespace created in step 2 (e.g. `my-reg-cred`).
. Go to `system-default-registry` and choose **⋮ > Edit Setting**.

> . Enter your registry's hostname and port (e.g. `registry.yourdomain.com:port`). Do not prefix the text with `http://` or `https://`.

Reviewer: ```suggestion
If your registry requires authentication (such as `.dockerconfigjson`), perform the following steps:

. Log in to Rancher and configure the default administrator password.
. Create a new registry secret in the `cattle-system` namespace using `kubectl`, Terraform, or the Rancher UI.
. If you create the secret outside the Rancher UI, ensure that xref:#_working_with_private_registry_credentials[backup labels are set].
. Select *☰ > Global Settings*.
. Find `system-default-registry-pull-secrets` and select *⋮ > Edit Setting*.
. Enter the name of the registry pull secret in the `cattle-system` namespace created in step 2 (such as `my-reg-cred`).
. Find `system-default-registry` and select *⋮ > Edit Setting*.
. Enter your registry hostname and port (such as `registry.yourdomain.com:port`). Do not prefix the text with `http://` or `https://`.
```

Reviewer: - [ ] Add xref for `xref:#_working_with_private_registry_credentials`

Author: I'm new to adoc, but the current xref works as is. Is your suggestion to add it as an xref entry in the header too?

Reviewer: @HarrisonWAffel Sorry should have provided context. Just need to update the formatting to align with our best practices. Updated via [ea442b9](https://github.com/rancher/rancher-product-docs/pull/1235/commits/ea442b9f20c6805d9d27bcb11884f4e03ef08d33).

> **Result:** Rancher propagates the configured pull secret into the required rancher-managed namespaces in the local cluster, and configures rancher-managed workloads to use that secret. Newly created and existing downstream clusters are automatically configured to point to the configured registry. All Rancher system images across all clusters are then pulled from your registry.

Reviewer: ```suggestion
*Result:* Rancher propagates the pull secret into the required Rancher-managed namespaces in the local cluster and configures Rancher-managed workloads to use it. Rancher automatically configures new and existing downstream clusters to point to the registry. All Rancher system images across all clusters then pull from your private registry.
```

[NOTE]
====

> For more detailed information on how Rancher works with *authenticated* default registries, view {xref-authenticated-private-registries}[this page]

Reviewer: ```suggestion
For details on how Rancher works with authenticated default registries, see {xref-authenticated-private-registries}[Authenticated Private Registries].
```

====

[IMPORTANT]
====

> Reconfiguring registry details at the global level may result in the redeployment of system components in the local cluster, and all downstream clusters which do not define their own cluster-level registry configurations. This may temporarily prevent access to downstream clusters via the Rancher UI.

Reviewer: ```suggestion
Reconfiguring registry details globally can trigger the redeployment of system components in the local cluster and downstream clusters that do not define cluster-level registry configurations. This can temporarily prevent access to downstream clusters through the Rancher UI.
```

====

[#_configure_a_cluster_level_private_registry]
== Configure a Cluster Level Private Registry

In addition to configuring a private registry at the global level, Rancher also supports defining cluster scoped registry configurations. To do so, you'll have to define the registry information in the advanced cluster options when you create a new cluster or update an existing cluster.

[IMPORTANT]
====
Reconfiguring registry details on an existing cluster may result in the redeployment of system-components in that cluster. This may temporarily prevent access to the cluster via the Rancher UI.
====

> 

Reviewer: ```suggestion
In addition to global registry configurations, Rancher supports cluster-scoped registry configurations. Specify registry details in *Advanced Options* when creating or updating a cluster.

[IMPORTANT]
====
Reconfiguring registry details on an existing cluster can trigger the redeployment of system components in that cluster. This can temporarily prevent access to the cluster through the Rancher UI.
====
```

[#_using_a_cluster_level_private_registry_with_provisioned_clusters]
=== Using a Cluster Level Private Registry with Provisioned Clusters

. Select **☰ > Cluster Management**.
. On the **Clusters** page, click **Create**.
. Choose a cluster type.
. In the **Cluster Configuration** go to the **Registries** tab.
. Check the box next to **Enable cluster scoped container registry for Rancher system container images**.
. Enter the registry hostname.
. Optional: Configure Credentials for the Cluster-Level Registry
. Under **Authentication** select **Create a HTTP Basic Auth Secret** and fill in the credential fields.
. Click **Create**.

**Result:** The new cluster pulls images from the private registry.

> This same process can also be followed when later updating the cluster configuration.

Reviewer: ```suggestion
. Select *☰ > Cluster Management*.
. On the *Clusters** page, click *Create*.
. Choose a cluster type.
. In *Cluster Configuration*, select the *Registries* tab.
. Select *Enable cluster-scoped container registry for Rancher system container images*.
. Enter the registry hostname.
. *Optional:* Under *Authentication*, select *Create an HTTP Basic Auth Secret* and enter the credentials.
. Click *Create*.

*Result:* The new cluster pulls images from the private registry.

Follow this same process when updating an existing cluster configuration.
```

. Under *Authentication* select *Create a HTTP Basic Auth Secret* and fill in the credential fields.
. Click *Create*.

*Result:* The new cluster pulls images from the private registry. This can be changed later by editing the cluster configuration.

[#_configure_a_cluster_level_private_registry_when_importing_clusters]
== Configure a Private Registry with Credentials when Importing a Cluster

. Select **☰ > Cluster Management**.
. On the **Clusters** page, click **Import Existing**.
. Choose a cluster type.
. Go to the **Registries** tab.
. Check the box next to **Enable cluster scoped container registry for Rancher system container images**.
. Enter the registry hostname.
. Optional: Select an optional image pull secret to authenticate with the registry

> . Click **Create**.

Reviewer: ```suggestion
*Result:* The new cluster pulls images from the private registry. You can change this setting later by editing the cluster configuration.

[#_configure_a_cluster_level_private_registry_when_importing_clusters]
== Configure a Private Registry with Credentials When Importing a Cluster

. Select *☰ > Cluster Management*.
. On the *Clusters* page, click *Import Existing*.
. Choose a cluster type.
. Select the *Registries* tab.
. Select *Enable cluster-scoped container registry for Rancher system container images*.
. Enter the registry hostname.
. *Optional:* Select an image pull secret to authenticate with the registry.
. Click *Create*.
```
